examples/parameter-loading/infection_manager.rs

```diff
@@ -0,0 +1,100 @@
+use rand_distr::Exp;
+
+use crate::InfectionStatus;
+use crate::InfectionStatusType;
+use crate::Parameters;
+
+define_rng!(InfectionRng);
+
+fn schedule_recovery(context: &mut Context, person_id: PersonId) {
+    let parameters = context.get_global_property_value(Parameters).clone();
+    let infection_duration = parameters.infection_duration;
+    let recovery_time = context.get_current_time()
+        + context.sample_distr(InfectionRng, Exp::new(1.0 / infection_duration).unwrap());
+    context.add_plan(recovery_time, move |context| {
+        context.set_person_property(person_id, InfectionStatusType, InfectionStatus::R);
+    });
+}
+
+fn handle_infection_status_change(context: &mut Context, event: PersonPropertyChangeEvent<InfectionStatusType>) {
+    if matches!(event.current, InfectionStatus::I) {
+        schedule_recovery(context, event.person_id);
+    }
+}
+
+pub fn init(context: &mut Context) {
+    context.subscribe_to_event(move |context, event:PersonPropertyChangeEvent<InfectionStatusType>| {
+        handle_infection_status_change(context, event);
+    });
+}
+
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::define_person_property_with_default;
+    use crate::define_person_property;
+    use ixa::context::Context;
+    use ixa::define_data_plugin;
+    use ixa::global_properties::ContextGlobalPropertiesExt;
+    use ixa::people::{ContextPeopleExt, PersonPropertyChangeEvent, PersonId};
+    use ixa::random::ContextRandomExt;
+    define_data_plugin!(RecoveryPlugin, usize, 0);
+
+    use crate::parameters_loader::ParametersValues;
+    
+    #[test]
+    fn test_handle_infection_change() {
+        let p_values = ParametersValues {
+            population: 10,
+            max_time: 10.0,
+            seed: 42,
+            foi: 0.15,
+            infection_duration: 5.0,
+            output_dir: ".".to_string(),
+            output_file: ".".to_string(),
+        };
+        let mut context = Context::new();
+
+        context.set_global_property_value(Parameters, p_values);
+        context.init_random(42);
+        init(&mut context);
+
+        context.subscribe_to_event(move |context, event:PersonPropertyChangeEvent<InfectionStatusType>| {
+            if matches!(event.current, InfectionStatus::R) {
+                *context.get_data_container_mut(RecoveryPlugin) += 1;
+            }
+        });
+
+        let population_size:usize = 10;
+        for _ in 0..population_size {
+            let person = context.add_person();
+
+            context.add_plan(1.0, move |context| {
+                println!("time {:?} - PersonID: {:?} - status: {:?}",
+                    context.get_current_time(), person,
+                    context.get_person_property(person, InfectionStatusType));
+                context.set_person_property(
+                    person,
+                    InfectionStatusType,
+                    InfectionStatus::I);
+            });
```

Comment:
Can someone help here? I am having a hard time with this test. The purpose of the test is to make sure that people who are set to be infectious will eventually recover (`handle_infection_status_change`). However, it isn't working properly. I added this print statement here and below to check people's infection status. Weird thing is, people actually recover if I include the statement from lines 79-82, but they never do if I remove that statement. What am I doing wrong? @jasonasher @ekr-cfa @k88hudson-cfa @ChiragKumar9

Reply:
I bet this is a consequence of lazy person properties. More later